### ISSUE TITLE
Normalize container CPU percentages

### ISSUE THREAD

**Issue by tombruijn** [appsignal/probes-rs#67]
Currently we report a container/cgroup with more than 1 CPU as 100% per CPU.
If a container has 2 CPUs, the max total CPU usage is 200%.

To normalize this percentage to a maximum of 100%, we need to know how many virtual CPUs there are set up as container limits.

## Local Docker container

Local container example: I specify it to have 2 CPUs and it returns 2.

Edit: This specifies which specific CPUs to use, which may not be entirely accurate to how many CPUs there are, because there's also the `--cpus` option.
```
$ docker run --rm -it --cpuset-cpus 2 ubuntu:22.04 bash
# cat /sys/fs/cgroup/cpuset.cpus
2
# cat /sys/fs/cgroup/cpuset.cpus.effective
2
```

More info on cgroup v2 spec: https://www.kernel.org/doc/html/latest/admin-guide/cgroup-v2.html
More info on Docker container resource limits: https://docs.docker.com/config/containers/resource_constraints/#cpu

## Amazon EC2?
We may also need to take AWS EC2 into consideration, because are they containers/cgroups? People say they aren't, but why is the `/sys/fs/` filesystem mounted then?

Example from an AWS EC2 t3.micro server:

```
# cat /sys/fs/cgroup/cpuset.cpus
<File not found>
$ cat /sys/fs/cgroup/cpuset.cpus.effective
0-1
```

Amazon lists the [t3.micro machine as having 2 vCPUs](https://aws.amazon.com/ec2/instance-types/t3/) (but it's throttled because it's a free tier). I read this output as having ~0 to 1 vCPU~ 2 CPUs, selecting the range of CPUs 0 and 1 (zero based index).

## Be warned!

Normalizing the CPU percentage in such a way is a backwards incompatible change. People relying on triggers for CPU usage of 500% will not be alerted of normalized 100% CPU usage after upgrading.

**Comment by tombruijn** [comment]
Related support conversation https://app.intercom.com/a/inbox/yzor8gyw/inbox/admin/4356044/conversation/16410700221387#part_id=comment-16410700221387-21909527416

**Comment by tombruijn** [comment]
There's a `/sys/fs/cgroup/cpu.max` file that shows interesting behavior.

> cpu.max
> A read-write two value file which exists on non-root cgroups. The default is "max 100000".
> 
> The maximum bandwidth limit. It's in the following format:
> 
>     $MAX $PERIOD
> 
> which indicates that the group may consume up to $MAX in each $PERIOD duration. "max" for $MAX indicates no limit. If only one number is written, $MAX is updated.

Some testing results:

```
docker run --rm -it --cpus "$CPUS" ubuntu:22.04 bash
```

- `$CPUS` == 4: `400000 100000` (4x)
- `$CPUS` == 0.5: `50000 100000` (0.5x)

Can we use the factor of the second value to determine how many cpus there are and use that to divide the current cpu usage?

```
400000 / 100000.0 = 4   CPUs
50000 / 100000.0  = 0.5 CPUs
```

Knowing how many CPUs there are we can say:

- 100% CPU usage / 1 CPUs = 100%
- 400% CPU usage / 4 CPUs = 100%
- 50% CPU usage / 0.5 CPUs = 100%

If the value is `max 100000` (the default when no container limits are set up, which is already a bad idea), we use the current behavior, which results in 400% CPU usage for 4 CPUs. We already have this warning on our [container host metrics docs page](https://docs.appsignal.com/metrics/host-metrics/containers.html#container-limits), so it should be fine:

> For container host metrics to be accurate, limits need to be set for every container.

Not done reading up, but best I got so far. The other `/sys/fs` files don't seem to really tell me a clear number of how many CPUs there are. At least not on every system, like AWS, or that's just because I tested it on a t3.micro and other non-micro machines report better values.
